### ISSUE TITLE
Support for autoscaling_group_tags in self-managed-node-group module

### ISSUE THREAD

**Issue by juniorz** [terraform-aws-modules/terraform-aws-eks#2072]
## Is your request related to a problem? Please describe.

`cluster-autoscaler` auto-discovery setup relies on tags placed in the ASG (see: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md#auto-discovery-setup).

However, the variable `tags` in the `self-managed-node-group` also applies the tags to `security_groups`, `iam_roles` and `instance_profile` - which does not make sense to have the same tags.

We can't also use `aws_autoscaling_group_tag` resource because they should not be combined with managing tags via `aws_autoscaling_group` directly (https://github.com/terraform-aws-modules/terraform-aws-eks/blob/d969e94bbea89ed616713bdab849f042ceda87ac/modules/self-managed-node-group/main.tf#L386).

## Describe the solution you'd like.

A `autoscaling_group_tags` variable, similarly to `iam_role_tags`, `launch_template_tags`, and `security_group_tags`.

**Comment by gferon** [comment]
@bryantbiggs if you think this is a good idea, I'm up for implementing it.

**Comment by bryantbiggs** [comment]
@gferon yes, please feel free to open a PR and I can take a look. Thank you!

**Comment by migueleliasweb** [comment]
Hi, this seems to also be affecting managed node groups as the tags, **when updated**, are not being propagated to the underlying ASG.

**Comment by antonbabenko** [comment]
This issue has been resolved in version 18.23.0 :tada:
